[IV-23-24] Objetivo 3

raultl12 opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

raultl12 committed:
doc: Creados criterios de eleccion closes #17

Se ha creado el archivo que contiene los criterios de eleccion que se deben seguir a la hora de incorporar una nueva herramienta al proyecto

raultl12 committed:
doc: elegido runtime closes #20

Se ha creado el archivo donde se discute la eleccion del runtime para el proyecto. La decision se ha hecho en base a los criterios establecidos

raultl12 committed:
doc: elegido gestor de tareas closes #18

Se ha elaborado el archivo donde se discute el gestor de tareas que se usará para el proyecto.

raultl12 committed:
doc: elegido gestor dependencias closes #19

Se ha añadido el archivo donde se discute el gestor de dependencias a usar en el proyecto.

raultl12 committed:
doc: elegidas herramientas de automatización closes #16

Se ha añadido el archivo donde se comentan todas las herramientas que serán usadas en el proyecto para la automatización de tareas

raultl12 committed:
doc: actualizado README closes #22

Se ha actualizado el readme para que contemple la nueva documentación sobre las herramientas seleccionadas para la automatización de operaciones

raultl12 committed:
doc: actualizado README con coprobación de sintaxis closes #23

Se ha añadido al README la información necesaria sobre la clase que va a ser comprobada asi como las instrucciones para llevar la comprobación a cabo

raultl12 committed:
feat: añadido iv.yaml closes #24

Se ha añadido el archivo iv.yaml con las claves necesarias para automatizar las tareas, el resto de claves ha sido copiado del pr del objetivo dos ya que no ha sido completado

raultl12 committed:
feat: archivo de scripts de velociraptor closes #25

Se ha creado el archivo velociraptor.ts donde se encuentra la orden para comprobar la sintaxis de la clase especificada en el README

raultl12 committed:
feat: añadido el codigo de clase usuario closes #26

Se ha añadido el codigo cuya sintaxsis sera comprobada, el codigo se ha copiado del pr del objetivo 2 por no estar este completo

raultl12 commented:
@JJ listo para revision

JJ commented on docs/eleccion_runtime.md:
```diff
@@ -0,0 +1,21 @@
+Es la opción mas popular y permite la ejecución asíncrona de javascript. No establece grandes criterios de seguridad y para ejecutar TypeScript debemos compilar previamente el código TypeScript a JavaScript. Es gratuito y open source. Importa modulos mediante CommonJS, lo que obliga a descargar/instalar dichos módulos antes de usarlos. 
+
+### Deno
+Deno surgió como una mejora más segura para Node.js. Fue creado por la misma persona que Node.js y cuenta con soporte nativo para TypeScript sin necesidad de compilar a JavaScript. Cuenta con herramientas propias como lintern y test runner. Además, también mejora la velocidad de ejecución frente a Node.js. Al igual que Node.js, es gratuito y de código abierto. En cuanto a los módulo, se importan  ...
```
porfa, en general, tratad sólo de poner la información que sea pertinente a la elección que hayáis hecho.

raultl12 replied:
resuelto en: https://github.com/raultl12/TeamFinder/pull/21/commits/eae46f18d42c0db750e94e1bc274c6b507d91efa

JJ commented on docs/eleccion_task_runner.md:
```diff
@@ -0,0 +1,23 @@
+### npm scripts
+Es el más usado con Node.js. Las tareas se escriben en la seccion scripts dentro del package.json. 
+
+### Gulp
```
¿Has mirado Snyk Advisor? ¿Qué puntuación tiene Gulp?
Aparte, tendrías que instalar node, ¿no?

raultl12 replied:
resuelto en: https://github.com/raultl12/TeamFinder/pull/21/commits/592ba43a5d591be569f84714622ea2f04ba912c8

JJ replied:
La mejor forma de resolverlo es eliminarlo, me temo. Con esa salud no debería estar aquí

JJ reviewed (changes requested):
Una vez que has elegido un runtime, no hace falta mencionar los que necesiten un runtime diferente para funcionar. Pero si los mencionas, por favor, que sean relevantes.
Aparte, también hay task runners como make o task que se pueden usar en cualquier lenguaje.

JJ commented:
Aparte de lo de arriba, velocirraptor tendrás que instalarlo en *tu* proyecto, así que habrá que hacer algún cambio en algún lado... También sería conveniente que pusieras un enlace, porque no es fácil de encontrar.

raultl12 committed:
doc: eliminada informacion irrelevante closes #28

Se ha eliminado la informacion que no era relevante sobre Deno en el archivo de eleccion de runtime

raultl12 committed:
doc: añadida informacion sobre Gulp closes #29

Se ha añadido informacion extra sobre Gulp en el fichero de eleccion de task runner

raultl12 committed:
doc: añadida informacion sobre velociraptor closes #30

Se ha ampliado la informacion de velociraptor añadiendo su pagina web

raultl12 committed:
package: Instalado velociraptor en el proyecto closes #27

Se ha instalado velociraptor en el proyecto para que pueda ser ejecutado desde él, sin necesidad de instalarlo globalmente

raultl12 commented:
@JJ Listo para revision

JJ commented on bin/vr:
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+# generated by deno install
```
¿Esto lo genera deno install aquí mismo? ¿no debería generarlo en algún path que estuviera en la variable $PATH?

raultl12 replied:
resuelto en: https://github.com/raultl12/TeamFinder/pull/21/commits/73280d80ca2412d90eab1e10582f46e144785a01

JJ commented on docs/eleccion_task_runner.md:
```diff
@@ -0,0 +1,23 @@
+### Gulp
+Es en task runner independiente en el que las tareas se escriben en un fichero .js. Se pueden incluir plugins externos. Utiliza node, por lo que hay que instalarlo. En Snyk Advisor no tiene una puntuación muy elevada(79/100).
+
+### Grunt
```
Sigue sin ser compatible con deno, porque necesita javascript. npm scripts tampoco. Y no has puesto herramientas generales como make.

raultl12 replied:
Resuelto en: https://github.com/raultl12/TeamFinder/pull/21/commits/420b8d7c9a0f165625380f26d19302d0289fdcaf y en https://github.com/raultl12/TeamFinder/pull/21/commits/a1d003c2133529b2149c8eb0c0e1531fdc1ccf5a

JJ reviewed (changes requested):
Sigue habiendo herramientas incompatibles con el runtime instalado.
Y también creo que deberías, al menos para la instalación de `vr`, usar `deno task`. Si no, no hay ningún elemento en tu proyecto que diga qué vas a instalar.

raultl12 committed:
delete: eliminado ejecutable de vr (#31)

Se ha eliminado el ejecutable de velociraptor porque estaba en los archivos del proyecto en lugrar de en /home/raul/.deno/bin:/home/raul/.local/bin:/home/raul/.deno/bin:/home/raul/.nvm/versions/node/v18.18.0/bin:/home/raul/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

raultl12 committed:
feat: instalacion de velociraptor closes #31

Se ha creado el archivo deno.json donde se especifica una tarea para deno task que instala velociraptor en una carpeta dentro de /home/raul/.deno/bin:/home/raul/.local/bin:/home/raul/.deno/bin:/home/raul/.nvm/versions/node/v18.18.0/bin:/home/raul/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

raultl12 committed:
doc: eliminados task runners incompatibles closes #32

Se han eliminado del archivo de eleccion de task runner aquellos que no eran compatibles con Deno(el runtime elegido)

raultl12 committed:
doc: añadidos task runners genericos closes #33

Se han añadido herramientas genéricas de task runner a la eleccion de task runner

raultl12 commented:
@JJ listo para revision

JJ commented on iv.yaml:
```diff
@@ -0,0 +1,6 @@
+lenguaje: TypeScript
+entidad: src/Usuario.ts
```
No coincide con el nombre del fichero. Por favor, compruébalo.

raultl12 replied:
Resuelto en: https://github.com/raultl12/TeamFinder/pull/21/commits/d694328d7d4aa3d043eba4312bee4169837ec2fd

JJ reviewed (approved):
:+1:

raultl12 committed:
doc: Arreglado el nombre de entidad closes #34

Se ha corregido la mayúscula en el nombre de la entidad del archivo iv.yaml
